compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/AMDAIEAssignNpuDmaBdIds.cpp

```diff
@@ -76,11 +76,12 @@ std::optional<uint32_t> getNumberIterations(scf::ForOp loop) {
 }
 
 /// Computes the number of BD IDs required between the current
-/// DMA copy operation and its corresponding DMA wait operation.
-/// If a sub-loop is encountered, it assumes that a BD ID expression will be
-/// used within the sub-loop. This approach ensures that the inner loop has
-/// access to a greater number of BD IDs, which is favorable for enabling
-/// efficient BD chaining in subsequent passes.
+/// DMA copy operation and its corresponding DMA wait operation; returns the DMA
+/// ops it traverses within same block and on same tile. If a sub-loop is
+/// encountered, it assumes that a BD ID expression will be used within the
```

Comment:
Could you point to where we make this assumption and explain this more?

Reply:
Actually [this comment](https://github.com/nod-ai/iree-amd-aie/blob/5319071511a701243680ff0be2068daeda03fd60/compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/AMDAIEAssignNpuDmaBdIds.cpp#L80-L81) seems to be related to [inner scf.for traversal here](https://github.com/nod-ai/iree-amd-aie/blob/5319071511a701243680ff0be2068daeda03fd60/compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/AMDAIEAssignNpuDmaBdIds.cpp#L127-L137).

I believe it was meant as a way to recursively traverse the inner loop and get corresponding BD ID requirement in case there are DMA ops operating on same tile as the "current DMA op".

Reply:
Oh, ok, I see that this is not new, I thought you changed this.